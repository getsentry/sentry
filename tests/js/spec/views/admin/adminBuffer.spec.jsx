import React from 'react';

import {mountWithTheme} from 'sentry-test/enzyme';

import AdminBuffer from 'app/views/admin/adminBuffer';

// TODO(dcramer): this doesnt really test anything as we need to
// mock the API Response/wait on it
describe('AdminBuffer', function() {
  describe('render()', function() {
    it('renders', function() {
      const wrapper = mountWithTheme(<AdminBuffer params={{}} />, {
        context: {
          router: TestStubs.router(),
        },
      });
      expect(wrapper).toSnapshot();
    });
  });
});
