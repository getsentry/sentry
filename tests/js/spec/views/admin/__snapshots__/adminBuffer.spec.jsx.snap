// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminBuffer render() renders 1`] = `
<AdminBuffer
  params={Object {}}
  resolution="1h"
>
  <div>
    <h3>
      Buffers
    </h3>
    <div
      className="box"
    >
      <div
        className="box-header"
      >
        <h4>
          About
        </h4>
      </div>
      <div
        className="box-content with-padding"
      >
        <p>
          Sentry buffers are responsible for making changes to cardinality counters — such as an issues event count — as well as updating attributes like
           
          <em>
            last seen
          </em>
          . These are flushed on a regularly interval, and are directly affected by the queue backlog.
        </p>
      </div>
    </div>
    <div
      className="box"
    >
      <div
        className="box-header"
      >
        <h4>
          Updates Processed
        </h4>
      </div>
      <withApi(InternalStatChart)
        label="Jobs"
        resolution="1h"
        since={1507603280}
        stat="jobs.finished.sentry.tasks.process_buffer.process_incr"
      >
        <InternalStatChart
          api={
            Client {
              "_chain": [Function],
              "_wrapRequest": [Function],
              "bulkUpdate": [Function],
              "handleRequestError": [Function],
              "hasProjectBeenRenamed": [Function],
            }
          }
          height={150}
          label="Jobs"
          resolution="1h"
          since={1507603280}
          stat="jobs.finished.sentry.tasks.process_buffer.process_incr"
        >
          <profiler(LoadingIndicator)>
            <Profiler
              disabled={false}
              includeRender={true}
              includeUpdates={false}
              name="LoadingIndicator"
              updateProps={Object {}}
            >
              <LoadingIndicator>
                <div
                  className="loading"
                >
                  <div
                    className="loading-indicator"
                    style={Object {}}
                  />
                  <div
                    className="loading-message"
                  />
                </div>
              </LoadingIndicator>
            </Profiler>
          </profiler(LoadingIndicator)>
        </InternalStatChart>
      </withApi(InternalStatChart)>
    </div>
    <div
      className="box"
    >
      <div
        className="box-header"
      >
        <h4>
          Revoked Updates
        </h4>
      </div>
      <withApi(InternalStatChart)
        label="Jobs"
        resolution="1h"
        since={1507603280}
        stat="buffer.revoked"
      >
        <InternalStatChart
          api={
            Client {
              "_chain": [Function],
              "_wrapRequest": [Function],
              "bulkUpdate": [Function],
              "handleRequestError": [Function],
              "hasProjectBeenRenamed": [Function],
            }
          }
          height={150}
          label="Jobs"
          resolution="1h"
          since={1507603280}
          stat="buffer.revoked"
        >
          <profiler(LoadingIndicator)>
            <Profiler
              disabled={false}
              includeRender={true}
              includeUpdates={false}
              name="LoadingIndicator"
              updateProps={Object {}}
            >
              <LoadingIndicator>
                <div
                  className="loading"
                >
                  <div
                    className="loading-indicator"
                    style={Object {}}
                  />
                  <div
                    className="loading-message"
                  />
                </div>
              </LoadingIndicator>
            </Profiler>
          </profiler(LoadingIndicator)>
        </InternalStatChart>
      </withApi(InternalStatChart)>
    </div>
  </div>
</AdminBuffer>
`;
